src/mlpack/bindings/R/mlpack/R/inline.R

```diff
@@ -0,0 +1,17 @@
+inlineCxxPlugin <- function(...) {
+    is_macos <- Sys.info()[["sysname"]] == "Darwin"
+    openmp_flag <- if (is_macos) "" else "$(SHLIB_OPENMP_CFLAGS)"
+    plugin <-
+        Rcpp::Rcpp.plugin.maker(
+            include.before = "#include <mlpack.h>",
+            libs           = paste(openmp_flag, "$(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)"),
+            LinkingTo      = c("RcppArmadillo", "Rcpp", "RcppEnsmallen", "mlpack"),
```

Comment:
Does `RcppEnsmallen` have any shared objects to link against?  ensmallen is header only, so at least in C++-land there's nothing to link against.  Anyway, I certainly don't know Rcpp plugins, so maybe this is needed just to indicate the dependency, I dunno. :smile:

Reply:
`LinkingTo` justs points to the header`s location.  So we need it.

Reply:
Much appreciate the explanation, thank you both! :+1:

Reply:
@eddelbuettel  speaking of awareness , [here](https://github.com/cgiachalis/mlpack.Rpkg) the demo package (template) I had created as part of https://github.com/mlpack/mlpack/issues/3613. Of course it works only if there is a plug-in helper but someone can follow the instructions in the issue if they can't wait for the CRAN release.